test/torchaudio_unittest/models/wav2vec2/huggingface_intergration_test.py

```diff
@@ -142,15 +144,22 @@ def test_import(self, config, _):
         for i, l in enumerate(output_lengths):
             self.assertEqual(ref[i, :l, ...], hyp[i, :l, ...])
 
-    @HF_CONFIGS
-    def test_recreate(self, config, factory_func):
-        """Imported models can be recreated via a factory function without Hugging Face transformers."""
-        imported = import_huggingface_model(self._get_model(config)).eval()
+    @PRETRAIN_CONFIGS
+    def test_import_pretrain(self, config, _):
+        """wav2vec2 models from HF transformers can be imported and yields the same results"""
+        original = self._get_model(config).eval()
+        imported = import_huggingface_model(original).eval()
+        self._test_import_pretrain(original, imported, config)
 
-        reloaded = factory_func(num_out=imported.encoder.readout.out_features)
-        reloaded.load_state_dict(imported.state_dict())
-        reloaded.eval()
+    @FINETUNE_CONFIGS
+    def test_import_finetune(self, config, _):
+        """wav2vec2 models from HF transformers can be imported and yields the same results"""
+        original = self._get_model(config).eval()
+        imported = import_huggingface_model(original).eval()
+        self._test_import_pretrain(original.wav2vec2, imported, config)
```

Comment:
for my understanding, is it the case that all finetune models only contains models of type `Wav2Vec2ForCTC`? and are pretrained models always `Wav2Vec2Model` or can it be either?

Reply:
My understanding is that, fine-tunes models are of `Wav2Vec2ForCTC` class.

For pretraining model, it seems that it can be either `Wav2Vec2Model` class or `Wav2Vec2ForCTC` class, and if it's later case, then it comes with an additional, untrained Liner module.

Reply:
@carolineechen Merging the PR, but feel free to continue on dicussion.

Reply:
yup np, the PR lgtm!